Lib/ufo2ft/__init__.py

```diff
@@ -12,7 +12,7 @@
 
 def compileOTF(ufo, outlineCompilerClass=OutlineOTFCompiler,
                featureCompilerClass=FeatureOTFCompiler, mtiFeaFiles=None,
-               kernWriter=KernFeatureWriter, markWriter=MarkFeatureWriter,
+               kernWriterClass=KernFeatureWriter, markWriterClass=MarkFeatureWriter,
```

Comment:
That makes sense. We can bump fontmake's minimum requirements for the next release, so I'm ok with API changes. The other major client of ufo2ft is Roboto, as you said, but they pinned the version in their requirements.txt, so it should be fine.
 
I'm ok if you also want to rename the `makeotfParts.py` module to something more meaningful like `featureCompiler.py`.